exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -373,9 +379,18 @@ private QueryProfile getQueryProfile(String queryId) {
   @GET
   @Path("/profiles/{queryid}.json")
   @Produces(MediaType.APPLICATION_JSON)
-  public String getProfileJSON(@PathParam("queryid") String queryId) {
+  public String getProfileJSON(@PathParam("queryid") String queryId,
+                               @Context HttpServletRequest req) {
     try {
-      return new String(work.getContext().getProfileStoreContext().getProfileStoreConfig().getSerializer().serialize(getQueryProfile(queryId)));
+      HttpSession session = req.getSession(false);
+      if (session == null || session.getAttribute(PROFILE_DATA) == null) {
```

Comment:
I think this session is not null here, because the `Cookie: JSESSIONID` is not null at this time. I recommend that :
```java
if(req.getSession().getAttribute(PROFILE_DATA) == null) { }
```

Reply:
if drill.exec.security.user.auth.enabled=false, session will be null